apis/v1alpha2/clusterapplicationresourcemapping_types.go

```diff
@@ -1,65 +1,62 @@
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// ClusterApplicationResourceMappingVersion defines the mapping for a specific version of an application resource.
+type ClusterApplicationResourceMappingVersion struct {
+	// Version is the version of the application resource that this mapping is for.
+	Version string `json:"version"`
+	// Containers is the collection of JSONPaths that container configuration may be written to.
+	Containers []string `json:"containers,omitempty"`
+	// Envs is the collection of JSONPaths that env configuration may be written to.
+	Envs []string `json:"envs,omitempty"`
+	// VolumeMounts is the collection of JSONPaths that volume mount configuration may be written to.
+	VolumeMounts []string `json:"volumeMounts,omitempty"`
+	// Volumes is the JSONPath that volume configuration must be written to.
+	Volumes string `json:"volumes"`
+}
 
 // ClusterApplicationResourceMappingSpec defines the desired state of ClusterApplicationResourceMapping
 type ClusterApplicationResourceMappingSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// Foo is an example field of ClusterApplicationResourceMapping. Edit clusterapplicationresourcemapping_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
-}
-
-// ClusterApplicationResourceMappingStatus defines the observed state of ClusterApplicationResourceMapping
-type ClusterApplicationResourceMappingStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Versions is the collection of versions for a given resource, with mappings.
+	Versions []ClusterApplicationResourceMappingVersion `json:"versions,omitempty"`
 }
 
-//+kubebuilder:object:root=true
-//+kubebuilder:subresource:status
-//+kubebuilder:resource:scope=Cluster
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:storageversion
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
 
 // ClusterApplicationResourceMapping is the Schema for the clusterapplicationresourcemappings API
 type ClusterApplicationResourceMapping struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   ClusterApplicationResourceMappingSpec   `json:"spec,omitempty"`
-	Status ClusterApplicationResourceMappingStatus `json:"status,omitempty"`
+	Spec ClusterApplicationResourceMappingSpec `json:"spec,omitempty"`
 }
 
-//+kubebuilder:object:root=true
+// +kubebuilder:object:root=true
 
 // ClusterApplicationResourceMappingList contains a list of ClusterApplicationResourceMapping
 type ClusterApplicationResourceMappingList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
-	Items           []ClusterApplicationResourceMapping `json:"items"`
-}
 
-func init() {
```

Comment:
we need to keep the init methods

Reply:
Reverted. f507ce0e9cb35a2d506b3da5ea25a20e87498103